server/paymentProviders/stripe/creditcard.ts

```diff
@@ -117,7 +117,12 @@ const createChargeAndTransactions = async (
   });
 
   const charge = paymentIntent.latest_charge || (paymentIntent as any).charges.data[0];
-  return createChargeTransactions(charge as Stripe.Charge, { order });
+  const transaction = await createChargeTransactions(charge as Stripe.Charge, { order });
+  if (order.SubscriptionId) {
+    const subscription = await models.Subscription.findByPk(order.SubscriptionId);
+    await subscription.update({ lastChargedAt: transaction.clearedAt });
+  }
```

Comment:
Seems like the subscription is not created yet at this point (I guess the same is true for other payment providers/options)

Maybe what is needed is something like the following in the `executeOrder` function in `server/lib/payments.ts` (line 867):

```typescript
order.interval && (await createSubscription(order, { lastChargedAt: transaction.clearedAt }));
```

Reply:
Nice catch. You're correct, for the first charge, we need to set lastChargeAt in execute order.